h2/src/main/org/h2/mvstore/db/MVTableEngine.java

```diff
@@ -266,17 +266,11 @@ public void initTransactions() {
         public void removeTemporaryMaps(BitSet objectIds) {
             for (String mapName : store.getMapNames()) {
                 if (mapName.startsWith("temp.")) {
-                    MVMap<?, ?> map = store.openMap(mapName);
-                    store.removeMap(map);
+                    store.removeMap(mapName);
                 } else if (mapName.startsWith("table.") || mapName.startsWith("index.")) {
                     int id = Integer.parseInt(mapName.substring(1 + mapName.indexOf('.')));
                     if (!objectIds.get(id)) {
-                        ValueDataType keyType = new ValueDataType(null, null, null);
-                        ValueDataType valueType = new ValueDataType(null, null, null);
-                        Transaction t = transactionStore.begin();
-                        TransactionMap<?, ?> m = t.openMap(mapName, keyType, valueType);
-                        transactionStore.removeMap(m);
-                        t.commit();
+                        store.removeMap(mapName);
```

Comment:
previously this was calling removeMap on transactionStore?

Reply:
Yes, but TransactionStore.openMap() call, which creates TransactionMap, just to be removed in the next line seems to be redundant. IMHO  createTempMap(), openMap(), openTempMap(), removeMap() all should go directly to MVStore, since they all operate with plain vanilla MVMap and has nothing to do with TransactionStore.

Reply:
I had an idea that we should just create a temporary MVStore for each session local temporary table, using whatever temporary filesystem the OS has, then we can just delete it when we're done, no need to do any kind of compaction on the main MVStore

Reply:
No, just an idea I am playing with at this point,